### ISSUE TITLE
CMake + Lists = 💔

### ISSUE THREAD

**Issue by mbrunnen** [ft/ufw#2]
I could not help but see that you had fun with cmake lists (e.g. `ufw_add_stringlist`) and I think it is due to the problem
with `ExternalProject_Add` and strings with semicolons, which are not lists? CMake wouldn't be cmake if there wouldn't be some whacky badly named parameter to fix this issue: Instead of `CMAKE_ARGS` use `CMAKE_CACHE_ARGS`:

So 
```cmake
ExternalProject_Add(
    ...
    CMAKE_ARGS
    -DZEPHYR_MODULES=a;b;c
    ...
```
would become:

```cmake
ExternalProject_Add(
    ...
    CMAKE_CACHE_ARGS
    -DZEPHYR_MODULES:STRING=a;b;c
    ...
```
Sorry, I am now too lazy for a proper PR, but I hope it still helps...

**Comment by ft** [comment]
Heya. I've completely forgotten about this. Sorry!

It's true, cmake is a terrible language. I've since replaced the recursive CMake calling scheme, to build large systems, by a `system` mode in `mmh`. This requires another external program, but fixes most of the technical and user-interface issues that the old recursive scheme had. That enabled removing all that terrible, non-maintainable mess of CMake code that was connected to that system.

Long story short: The codepaths, that use the `ExternalProject` CMake module don't exist anymore. Therefore this issue is void now.
